### PR TITLE
Added a new Option JSON Field in the Convert JSON to CSV Action, and …

### PR DESCRIPTION
**Pull request by cs-adarsh-gupta**
#### UTCs:

 

- [x] Connector installation verified.

- [x] Connector logo verified.

- [x] Actions and Playbook list verified.

- [x] Playbook tags verified.

- [x] All Playbook are in info mode verified.

- [x] All Playbook are in inactive mode verified.

Here are some Performance Numbers on FSR VM with Same Environment, Same CSV File with 20k+ entries.
Polars is able to perform faster (almost half the time) and uses less CPU Memory.

1 1 - a
1 2 - a
log1 1
log1 2

### CODE REVIEW

File: csv-data-management/extract_data_from_csv.py [hunk 1]
```diff
@@ -116,41 +123,42 @@ def merge_two_csv_and_extract_data(config, params):
...
         try:
             if params.get('deDupValuesOn'):
                 deDupValuesOn = params.get('deDupValuesOn')
                 deDupValuesOn = deDupValuesOn.split(",")
-                combined_recordSet = combined_recordSet.drop_duplicates(subset=deDupValuesOn, keep='first')
-        except Exception as err:
-            logger.error('Error in deduplicating data  extract_data_from_csv(): %s' % str(err))
-            raise ConnectorError('Error in deduplicating data  extract_data_from_csv(): %s' % str(err))
+                combined_recordSet = combined_recordSet.unique(subset=deDupValuesOn, keep='first')
+        except Exception as Err:
```
Comment (cs-pooja-singh) [inline csv-data-management/extract_data_from_csv.py:134]: why variable names with uppercase?

File: csv-data-management/extract_data_from_csv.py [hunk 2]
```diff
@@ -10,28 +10,29 @@
 from uuid import uuid4
 import requests
 import pandas as pd
```
Comment (cs-pooja-singh) [inline csv-data-management/extract_data_from_csv.py:12]: do you still need pandas?

File: csv-data-management/extract_data_from_csv.py [hunk 3]
```diff
@@ -456,11 +499,11 @@ def _ds_filter(params, ds):
         if input_type == 'On Values Matching a Regex':
             reg = params.get('filter')
             columnName = params.get('filterColumnName')
-            df = df[df[columnName].str.match(reg) == True]
+            df = df.filter(pl.col(columnName).str.contains(reg))
```
Comment (cs-pooja-singh) [inline csv-data-management/extract_data_from_csv.py:500]: while we are at it, could you cleanup the variable reinitialization at line 496?

File: csv-data-management/extract_data_from_csv.py [hunk 4]
```diff
@@ -470,34 +513,41 @@ def _df_to_csv(df, filename=None):
...
-        df.to_csv('/tmp/{}'.format(file_name.split(".")[0]) + '.zip', encoding='utf-8', header='true',
-                  compression=compression, index=False)
-        filepath = '/tmp/{}'.format(file_name.split(".")[0]) + '.zip'
+
+        if isinstance(df, pd.DataFrame):
+            df.to_csv('/tmp/{}'.format(file_name.split(".")[0]) + '.zip', encoding='utf-8', header='true',
+                      compression=compression, index=False)
+            filepath = '/tmp/{}'.format(file_name.split(".")[0]) + '.zip'
+
+        else:
```
Comment (cs-pooja-singh) [inline csv-data-management/extract_data_from_csv.py:520]: what exactly is df in this case?

Comment (cs-pooja-singh) [reply csv-data-management/extract_data_from_csv.py:520]: also, why no compression in this case?

File: csv-data-management/info.json [hunk 5]
```diff
@@ -772,30 +772,105 @@
           "editable": true,
           "value": "",
           "options": [
+            "JSON Field",
```
Comment (cs-pooja-singh) [inline csv-data-management/info.json]: call it "JSON" instead of "JSON Field"

Comment (cs-pooja-singh) [reply csv-data-management/info.json]: also, change the title from "Type" to "Input Type"

File: csv-data-management/info.json [hunk 6]
```diff
@@ -772,30 +772,105 @@
...
+                "tooltip": "Provide name for resulant CSV file. This file will be compressed"
+              },
+              {
+                "title": "Record Path",
+                "type": "text",
+                "name": "record_path",
+                "required": false,
+                "visible": true,
+                "editable": true,
+                "value": "",
```
Comment (cs-pooja-singh) [inline csv-data-management/info.json]: give a placeholder to understand what format of input you are expecting

File: csv-data-management/info.json [hunk 7]
```diff
@@ -772,30 +772,105 @@
...
+                "tooltip": "Provide the JSON data that you want to change to CSV"
+              },
+              {
+                "title": "Provide CSV Filename",
+                "type": "text",
+                "name": "csvFileName",
+                "required": true,
+                "visible": true,
+                "editable": true,
+                "value": "",
```
Comment (cs-pooja-singh) [inline csv-data-management/info.json]: better to omit the key itself instead of providing an empty value

File: csv-data-management/info.json [hunk 8]
```diff
@@ -772,30 +772,105 @@
...
+                "required": true,
+                "visible": true,
+                "editable": true,
+                "value": ""
+              },
+              {
+                "title": "Provide CSV Filename",
+                "type": "text",
+                "name": "csvFileName",
+                "required": true,
```
Comment (cs-pooja-singh) [inline csv-data-management/info.json:795]: why is this required? can we not use the name of the file from the attachment itself?

File: csv-data-management/requirements.txt [hunk 9]
```diff
@@ -1,3 +1,5 @@
 # Add Python Dependencies
 pandas
 numpy
+polars
+pyarrow
```
Comment (cs-pooja-singh) [inline csv-data-management/requirements.txt]: where do you use this one? I don't see related import statements

File: csv-data-management/extract_data_from_csv.py (hunk 2 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/extract_data_from_csv.py:12]: > do you still need pandas?
For the Action, Convert JSON to CSV, JSON Field. There is no polars function which can linearize Nested JSON Fields having a record path and Meta variables, It would simply give an error. So I had to use pandas.json_normalize function at that action only. If it's given that there will be no nested data within the JSON provided, I can change it to polars.

File: csv-data-management/extract_data_from_csv.py (hunk 3 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/extract_data_from_csv.py:500]: > while we are at it, could you cleanup the variable reinitialization at line 496?
Yes, Done.

File: csv-data-management/extract_data_from_csv.py (hunk 4 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/extract_data_from_csv.py:520]: Here, Dataframe can be of pandas and polars (Pandas df only in case of Convert JSON to CSV, JSON Field action) , As there was no direct way of Compression in Polars write_csv action, I had to make two conditions if the DF is of Pandas or Polars.

File: csv-data-management/info.json (hunk 5 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/info.json]: Yes, Done.

File: csv-data-management/info.json (hunk 7 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/info.json]: Done.

File: csv-data-management/requirements.txt (hunk 9 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/requirements.txt:5]: > where do you use this one? I don't see related import statements
Pyarrow is being internally used by Polars for some functionalities.

File: csv-data-management/info.json (hunk 6 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/info.json]: > give a placeholder to understand what format of input you are expecting

Done

File: csv-data-management/requirements.txt (hunk 9 above)
Comment (cs-adarsh-gupta) [reply csv-data-management/requirements.txt]: > where do you use this one? I don't see related import statements
I was using different approach at the time of importing this, But now it's not necessary. I have removed this.

**Review by cs-prerna-joshi — APPROVED** [review]
